mlir/utils/performance/analysis/compileAndCollectTuningData.py

```
@@ -0,0 +1,604 @@
+          flush=True)
+    if current == total:
+        print()  # New line when complete
+
+def main():
+    """Main function to process configurations and collect tuning data."""
+    parser = argparse.ArgumentParser(
+        description="Compile configurations and collect tuning data",
+        formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument('--op', required=True,
+                        help='Operation to perform (e.g., "compile")',
+                        choices=['conv', 'gemm', 'attention'])
+    parser.add_argument('config_tsv', help='Path to the tuning database file')
+    
+    args = parser.parse_args()
+
+    # Get the paths to the rocmlir binaries
+    build_bin_dir = os.path.dirname(os.path.abspath(__file__))
+    rocmlir_root = os.path.dirname(build_bin_dir)
+    paths = perfRunner.create_paths(None, rocmlir_root)
+
+    # Parse the configuration file
+    configs = perfRunner.read_tuning_db(args.config_tsv, True)
+    op_configs = None
+    if (args.op == 'conv'):
+        op_configs = perfRunner.getConvConfigurations(args.config_tsv)
+    elif (args.op == 'gemm'):
+        op_configs = perfRunner.getGemmConfigurations(args.config_tsv)
+    elif (args.op == 'attention'):
+        op_configs = perfRunner.getAttentionConfigurations(args.config_tsv)
+    else:
+        print(f"Error: Unknown operation '{args.op}'")
+        sys.exit(1)
+    print(f"Found {len(configs)} configurations to process")
+    
+    # Process each configuration
+    results = []
+    total_configs = len(configs)
+    for i, (config, test_args) in enumerate(zip(configs, op_configs)):
+        print_progress(i, total_configs)
+        metrics = compile_and_collect_data(config, configs[config], test_args,
+                                           args.op, paths)
+        results.append(metrics)
+
+    #Write the results to a final tsv file
+    write_results_to_tsv(results, configs)
+
+if __name__ == "__main__":
+    main()
```

If we plan to use this script as part of CI (since it is added to ci-performance-scripts) it would be good to pass main's return value so that we can ensure that CI would fail if we had non-zero return value at some point. (we had a lot of problems previously with different python scripts which weren't propagating errors properly, which led to silent failures in CI)

```suggestion
    sys.exit(main())
```